src/main/kotlin/org/jitsi/nlj/rtp/bandwidthestimation/GoogleCcEstimator.kt

```diff
@@ -0,0 +1,113 @@
+package org.jitsi.nlj.rtp.bandwidthestimation
+
+import java.time.Duration
+import java.time.Instant
+import kotlin.properties.Delegates
+import org.jitsi.nlj.stats.NodeStatsBlock
+import org.jitsi.nlj.util.Bandwidth
+import org.jitsi.nlj.util.DataSize
+import org.jitsi.nlj.util.bps
+import org.jitsi.nlj.util.createChildLogger
+import org.jitsi.nlj.util.kbps
+import org.jitsi.nlj.util.mbps
+import org.jitsi.utils.logging.DiagnosticContext
+import org.jitsi.utils.logging2.Logger
+import org.jitsi_modified.impl.neomedia.rtp.remotebitrateestimator.RemoteBitrateEstimatorAbsSendTime
+import org.jitsi_modified.impl.neomedia.rtp.sendsidebandwidthestimation.SendSideBandwidthEstimation
+
+private val defaultInitBw: Bandwidth = 2.5.mbps
+private val defaultMinBw: Bandwidth = 30.kbps
+private val defaultMaxBw: Bandwidth = 20.mbps
+
+class GoogleCcEstimator(diagnosticContext: DiagnosticContext, parentLogger: Logger) :
+    BandwidthEstimator(diagnosticContext) {
+    override val algorithmName = "Google CC"
+
+    /* TODO: Use configuration service to set this default value. */
+    override var initBw: Bandwidth = defaultInitBw
+    /* TODO: observable which sets the components' values if we're in initial state. */
+
+    override var minBw: Bandwidth by Delegates.observable(defaultMinBw) {
+        _, _, newValue ->
+        bitrateEstimatorAbsSendTime.setMinBitrate(newValue.bps.toInt())
+        sendSideBandwidthEstimation.setMinMaxBitrate(newValue.bps.toInt(), maxBw.bps.toInt())
+    }
+
+    override var maxBw: Bandwidth by Delegates.observable(defaultMaxBw) {
+        _, _, newValue ->
+        sendSideBandwidthEstimation.setMinMaxBitrate(minBw.bps.toInt(), newValue.bps.toInt())
+    }
+
+    private val logger = parentLogger.createChildLogger(GoogleCcEstimator::class)
+
+    /**
+     * Implements the delay-based part of Google CC.
+     */
+    private val bitrateEstimatorAbsSendTime = RemoteBitrateEstimatorAbsSendTime(diagnosticContext, logger)
+    init {
+        bitrateEstimatorAbsSendTime.setMinBitrate(minBw.bps.toInt())
+    }
+
+    /**
+     * Implements the loss-based part of Google CC.
+     */
+    private val sendSideBandwidthEstimation = SendSideBandwidthEstimation(diagnosticContext, initBw.bps.toLong(), logger)
+    init {
+        sendSideBandwidthEstimation.setMinMaxBitrate(minBw.bps.toInt(), maxBw.bps.toInt())
```

Comment:
oh, looks like we have 2 `init` blocks here...i don't think we actually need 2 though

Reply:
I was trying to group each init block with the initialization-time object it was modifying.

Reply:
thinking about this some more, i think we could actually get rid of the need for this by doing

```
private val sendSideBandwidthEstimation = SendSideBandwidthEstimation(diagnosticContext, initBw.bps.toLong(), logger).also { setMinMaxBitrate(minBw.bps.toInt(), maxBw.bps.toInt()) }
```

and taking care of it all in the initialization